### PR TITLE
Split config.h into multiple files

### PR DESCRIPTION
**Pull request by Arceveti**

### CODE REVIEW

File: include/config/config_camera.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (thecozies) [inline include/config/config_camera.h:1]:
```suggestion
#pragma audio_conf
```

File: include/config/config_graphics.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#pragma once
```
Comment (thecozies) [inline include/config/config_graphics.h:1]:
```suggestion
#pragma graphics_conf
```

File: include/config/config_graphics.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+// Whenever you change this, make sure to run "make -C tools clean" to rebuild the skybox tool (alternatively go into skyconv.c and change the file in any way (like adding/deleting a space) to specifically rebuild that tool).
+// When increasing this, you should probably also increase the GFX pool size. (the GFX_POOL_SIZE define in src/game/game_init.h)
+#define SKYBOX_SIZE 1
+
+// When this option is enabled, LODs will ONLY work on console.
+// When this option is disabled, LODs will work regardless of whether console or emulator is used.
+// Regardless of whether this setting is enabled or not, you can use gIsConsole to wrap your own code in a console check.
+#define AUTO_LOD
+
+// Disable AA (Recommended: it changes nothing on emulator, and it makes console run better)
```
Comment (thecozies) [inline include/config/config_graphics.h:23]:
```suggestion
// Disable AA (Improves console performance but causes visible seams between unconnected geometry)
```
@Reonu 😈

File: include/config/config_menu.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#pragma once
```
Comment (thecozies) [inline include/config/config_menu.h:1]:
```suggestion
#pragma menu_conf
```

File: include/config/config_movement.h [hunk 5]
```diff
@@ -0,0 +1,78 @@
+#pragma once
```
Comment (thecozies) [inline include/config/config_movement.h:1]:
```suggestion
#pragma movement_conf
```

File: include/config/config_goddard.h [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+#define GODDARD_EASTER_EGG
+
+// Disables the demo that plays when idle on the start screen (has no effect if KEEP_MARIO_HEAD is disabled)
+#define DISABLE_DEMO
+
+// -- Compatibility safeguards. Don't mess with these unless you know what you're doing.--
+#ifndef KEEP_MARIO_HEAD
+#undef GODDARD_EASTER_EGG
+#define DISABLE_DEMO
+#endif // !KEEP_MARIO_HEAD
```
Comment (thecozies) [inline include/config/config_goddard.h:20]: 💯

File: include/config/config_debug.h [hunk 7]
```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+/******************
+ * DEBUG SETTINGS *
+ ******************/
+
```
[the remark this answers is not in the document]
Comment (Arceveti) [reply include/config/config_debug.h:6]: What is that supposed to do? Prevent accidentally leaving debug stuff on?

**Pull request closed without merging**